### DISCUSSION TITLE
State of Multiplayer

### DISCUSSION THREAD

**Discussion by ProPeach** [tsunamayo/Starship-EVO#3958] [General]
Hey Tsuna o/
Recently there have been a lot of posts on the Steam forums asking about multiplayer, as well as lots of newer players on Discord asking about the same thing. As it stands, multiplayer is not very accessible as it requires a port forward connection, and is prone to desyncs and is generally unstable.

Do you have any idea of when you'll have time to work on the multiplayer experience? I don't mean to drag your attention away from your near term goals, especially after such a huge success as novox, but it does seem that multiplayer is at the front of many players minds, and is holding a lot of players back from playing the game more.

Happy to hear any and all thoughts you have on this Tsuna!

**Comment by PopkaAkula** [comment]
Multiplayer is a key feature for this game and I believe that always having it on the back-burner will be detrimental to the game and it's community. Just hearing occasional news about your plans for multiplayer would be enough for a lot of people.

**Comment by Garrett-C** [comment]
I also agree that multiplayer is something that would be beneficial to the game in the near future. I think the ability for people to properly play in multiplayer allows them to create their own gameplay which in turn gives you time to flesh out the proper gameplay loops. 

Currently all there is to the game is single-player building. With MP there is single-player building, multiplayer building, multiplayer fights etc. When players can play together there are a lot of opportunities to play the game more, even with just the creative element.

**Comment by Vicomt** [comment]
I would also like to join the chorus (with Oojimalfip, my bro and companion builder)
We've been waiting patiently while you rework various things, it's good, it needs doing.
But please. stabilise MP. that's all we want.

**Comment by Jorpis-Jorah** [comment]
I'm just getting HYPED UP lol

**Comment by tsunamayo** [comment]
Sorry I am just seeing this. Yes I havent given up on MP - it is just that I am massively time constrained.
Actually I have been switching the network tech at the beg of the year, so I am now able to improve it. 
I want to try to use the Steam network layer to improve the connectivity. Also yes desyncs are an issue, if you have some clear test case please share them.

**Reply by ProPeach** [reply]
Thanks for replying! I remembered you were waiting for the new framework before switching, I think that's why we stopped making mp based reports. Steam integrated mp would be *fantastic*, I'll see if I can get some reliable reports going

**Comment by Jorpis-Jorah** [comment]
I ask you...PLEASE don’t let the Mp be a clone of Space Engineer’s Mp, I’ve been joining servers without mods in them and the multiplayer is very laggy and problematic, it’s like Sc’s Mp framework is about to break its back and pas out. So Please, when the time comes I hope you can stabilise the Mp unlike Sc
